packages/yourloops/pages/profile/patient-form.tsx

```diff
@@ -26,115 +26,170 @@
-        label={t('birthdate')}
-        value={birthDate}
-        onChange={event => setBirthDate(event.target.value)}
-        error={errors.birthDate}
-        helperText={errors.birthDate && t('required-field')}
-        className={classes.formInput}
-      />
-      <TextField
-        id="profile-textfield-birthplace"
-        label={t('birthplace')}
-        value={birthPlace}
-        onChange={event => setBirthPlace(event.target.value)}
-        className={classes.formInput}
-        inputProps={{ maxlength: '50' }}
-      />
-      <FormControl className={`${props.classes.formInput}`}>
-        <InputLabel id="profile-select-gender-label" htmlFor="profile-select-gender">{t('gender')}</InputLabel>
-        <Select
-          id="profile-select-gender"
-          labelId="profile-select-gender-label"
-          value={sex}
-          onChange={event => setSex(event.target.value as string)}
-        >
-          <MenuItem value="" aria-label={t('none')}>{t('none')}</MenuItem>
-          <MenuItem value="M" aria-label={t('male')}>{t('male')}</MenuItem>
-          <MenuItem value="F" aria-label={t('female')}>{t('female')}</MenuItem>
-        </Select>
-      </FormControl>
-      <TextField
-        id="profile-textfield-referring-doctor"
-        label={t('referring-doctor')}
-        value={referringDoctor}
-        onChange={event => setReferringDoctor(event.target.value)}
-        className={classes.formInput}
-        inputProps={{ maxlength: '50' }}
-      />
-      {country === 'FR' &&
-      <>
+      <Box className={classes.inputContainer}>
         <TextField
-          id="profile-textfield-ins"
-          label={t('ins')}
-          value={ins}
-          onChange={event => setIns(event.target.value)}
+          id="profile-textfield-birthdate"
+          label={t('birthdate')}
+          value={profileForm.birthday}
+          onChange={event => updateProfileForm(ProfileFormKey.birthday, event.target.value)}
+          error={errors.birthday}
+          helperText={errors.birthday && t('required-field')}
           className={classes.formInput}
-          inputProps={{ maxlength: '15' }}
-          error={errors.ins}
-          helperText={errors.ins && t('field-with-exactly-15-characters')}
+          inputProps={{ maxLength: '10' }}
         />
         <TextField
-          id="profile-textfield-ssn"
-          label={t('ssn')}
-          value={ssn}
-          onChange={event => setSsn(event.target.value)}
+          id="profile-textfield-birthplace"
+          label={t('birthplace')}
+          value={profileForm.birthPlace}
+          onChange={event => updateProfileForm(ProfileFormKey.birthPlace, event.target.value)}
           className={classes.formInput}
-          inputProps={{ maxlength: '15' }}
-          error={errors.ssn}
-          helperText={errors.ssn && t('field-with-exactly-15-characters')}
+          inputProps={{ maxLength: '50' }}
         />
-      </>
+      </Box>
+      {country === 'FR' &&
```

Comment:
A good practice would be to make this string less prone to change, I see 2 solutions:
- define this string `'FR'` at the top of the file
```ts
const FR_COUNTRY_CODE = 'FR'
```
- have it defined in an enum containing multiple country codes
```ts
export enum CountryCode = {
  // ... Other relevant countries
  France = 'FR'
}
```

Reply:
Ok for an Enum, it's a nice solution !